contracts/src/Payer.sol

```diff
@@ -0,0 +1,1045 @@
+        __Pausable_init();
+
+        PayerStorage storage $ = _getPayerStorage();
+
+        $.minimumRegistrationAmountMicroDollars = DEFAULT_MINIMUM_REGISTRATION_AMOUNT_MICRO_DOLLARS;
+        $.minimumDepositAmountMicroDollars = DEFAULT_MINIMUM_DEPOSIT_AMOUNT_MICRO_DOLLARS;
+        $.withdrawalLockPeriod = DEFAULT_WITHDRAWAL_LOCK_PERIOD;
+        $.maxTolerableDebtAmountMicroDollars = DEFAULT_MAX_TOLERABLE_DEBT_AMOUNT_MICRO_DOLLARS;
+        $.transferFeesPeriod = DEFAULT_MINIMUM_TRANSFER_FEES_PERIOD;
+
+        _setUsdcTokenContract(usdcToken);
+        _setNodesContract(nodesContract);
+
+        require(_grantRole(DEFAULT_ADMIN_ROLE, initialAdmin), FailedToGrantRole(DEFAULT_ADMIN_ROLE, initialAdmin));
+        require(_grantRole(ADMIN_ROLE, initialAdmin), FailedToGrantRole(ADMIN_ROLE, initialAdmin));
+    }
+
+    /* ============ Payers Management ============ */
+
+    /**
+     * @inheritdoc IPayer
+     */
+    function register(uint64 amount) external whenNotPaused {
+        PayerStorage storage $ = _getPayerStorage();
+
+        require(amount >= $.minimumRegistrationAmountMicroDollars, InsufficientAmount());
+
+        if (_payerExists(msg.sender)) revert PayerAlreadyRegistered();
+
+        $.usdcToken.safeTransferFrom(msg.sender, address(this), amount);
+
+        // New payer registration
+        $.payers[msg.sender] = Payer({
+            balance: amount,
+            debtAmount: 0,
+            creationTimestamp: uint64(block.timestamp),
+            latestDepositTimestamp: uint64(block.timestamp),
+            latestDonationTimestamp: 0,
+            isActive: true
+        });
+
+        // Add new payer to active and total payers sets
+        require($.activePayers.add(msg.sender), FailedToRegisterPayer());
+        require($.totalPayers.add(msg.sender), FailedToRegisterPayer());
+
+        _increaseTotalAmountDeposited(amount);
+
+        emit PayerRegistered(msg.sender, amount);
+    }
+
+    /**
+     * @inheritdoc IPayer
+     */
+    function deposit(uint64 amount) external whenNotPaused nonReentrant onlyPayer(msg.sender) {
+        _validateAndProcessDeposit(msg.sender, msg.sender, amount);
+    }
+
+    /**
+     * @inheritdoc IPayer
+     */
+    function donate(address payer, uint64 amount) external whenNotPaused {
+        _revertIfPayerDoesNotExist(payer);
+
+        _validateAndProcessDeposit(msg.sender, payer, amount);
+        PayerStorage storage $ = _getPayerStorage();
+
+        $.payers[payer].latestDonationTimestamp = uint64(block.timestamp);
+
+        emit Donation(msg.sender, payer, amount);
+    }
+
+    /**
+     * @inheritdoc IPayer
+     */
+    function deactivatePayer(uint256 nodeId, address payer) external whenNotPaused onlyNodeOperator(nodeId) {
+        _revertIfPayerDoesNotExist(payer);
+
+        _deactivatePayer(nodeId, payer);
+    }
+
+    /**
+     * @inheritdoc IPayer
+     */
+    function deletePayer(address payer) external whenNotPaused onlyRole(ADMIN_ROLE) {
```

Comment:
Why do wee need to delete payers? Isn't deactivation more than sufficient for our on-chain needs?